### PR TITLE
Add workflow for bumping version

### PR DESCRIPTION
**Pull request by cmoog**

### CODE REVIEW

**Comment by cmoog** [comment]
@deansheather my current thinking is that the `cdr/coder-cli` release workflow will invoke this workflow... but I'm not too happy with how that plan is shaping up.

**Comment by deansheather** [comment]
Is there no way to do this from within the coder-cli release itself? Could just give coder-cli an SSH key with permission to do this stuff

**Comment by cmoog** [comment]
Yes that is an option, we could move this workflow code to coder-cli. I do think the additional authentication complexity is a considerable downside to that approach though. I *think* we'd also need to create a PAT for `gh`

**Comment by deansheather** [comment]
You'd still need code in coder-cli either way, so if you put the whole thing in coder-cli then you avoid having code in two places

**Comment by cmoog** [comment]
The template living in a separate repo is arguably worse.

File: .github/workflows/bump-coder-cli.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            ./ci/scripts/gen-coder-cli-formula.sh ${{ github.event.inputs.version }}
+            git diff
+            git checkout -b bump-coder-cli-${{ github.event.inputs.version }}
+            git add coder-cli.rb
+            git commit -m "feat: bump coder-cli version to v${{ github.event.inputs.version }}"
+            git push --set-upstream origin bump-coder-cli-${{ github.event.inputs.version }}
+            gh pr create \
+              --title="bump coder-cli version to v${{ github.event.inputs.version }}" \
+              --body="" \
+              --reviewer cmoog
```
Comment (deansheather) [inline .github/workflows/bump-coder-cli.yaml:31]: Move this to a separate script file.

**Review by deansheather — APPROVED** [review]
Looks good to me.

**Comment by cmoog** [comment]
This has gotten a bit stale. Closing for now. We should first consider using something like https://github.com/goreleaser/goreleaser

**Pull request closed without merging**